langlib/langlib-test/src/test/resources/test-src/type-param/type_param_test_negative.bal

```diff
@@ -47,8 +47,8 @@ function testError1() {
     FooError f = error("Foo");
     boolean f1 = f.message();    // incompatible types: expected 'boolean', found 'string'
     string f2 = f.message(); // No error;
-    Foo x1 = f.message(); // incompatible types: expected 'Foo', found 'string'
-    Bar x2 = f.message(); // incompatible types: expected 'Bar', found 'Foo'
+    Foo x1 = f.message(); // incompatible types: expected '"Foo"', found 'string'
+    Bar x2 = f.message(); // incompatible types: expected '"Bar"', found 'Foo'
```

Comment:
I think the comment is wrong, this should print `incompatible types: expected '"Bar"', found 'string'`

Shall we add a test case where name of the constant differs from the constant value. IMO the `expected` should be the type-name not the content (when there is a type name).

Reply:
Will look into this.

Reply:
It shows the content, not the type-name. As an example
```bal
const Aaa = "Foo";
const Bbb = "Bar";
type FooError error;

function testError1() {
    FooError f = error("Foo");
    boolean f1 = f.message();
    Aaa x1 = f.message();
    Bbb x2 = f.message();
}
```
 will show
```
Compiling source
        ex-finite.bal
ERROR [ex-finite.bal:(7:18,7:29)] incompatible types: expected 'boolean', found 'string'
ERROR [ex-finite.bal:(8:14,8:25)] incompatible types: expected '"Foo"', found 'string'
ERROR [ex-finite.bal:(9:14,9:25)] incompatible types: expected '"Bar"', found 'string'
error: compilation contains errors
```

Reply:
I'm not sure if it should do that, we usually display the type reference in error messages.
```ballerina
type A int;

public function main() {
    A a = "hello"; // incompatible types: expected 'A', found 'string'(BCE2066)
}
```

Shall we raise this in slack dev-compiler channel?

Reply:
Can't the above behaviour is correct because it is declared as a `const` and not as a type, or may I need to raise this in slack dev-compiler channel?
```bal
type A int;
const fruit = "Orange";
const numval = 2;
type unionType "Apple"|"Mango";
const unionconst = "Apple"|"Mango"; //type is required for constants with expressions(BCE3500)

public function main() {
    A a = "hello"; // incompatible types: expected 'A', found 'string'(BCE2066)
    fruit x1 = 4; // incompatible types: expected 'Orange', found 'int'(BCE2066)
    numval x2 = "aa"; // incompatible types: expected '2', found 'string'(BCE2066)
    unionType x3 = 0; // incompatible types: expected 'unionType', found 'int'(BCE2066)
    "number" num = 2; // incompatible types: expected 'number', found 'int'(BCE2066)
}
```

Reply:
According to the [slack discussion](https://ballerina-platform.slack.com/archives/CAXS256RX/p1641965098001400), it was decided that the error message should display the constant name there. So created issue https://github.com/ballerina-platform/ballerina-lang/issues/34640 for this.